lib/utils.js

```diff
@@ -41,14 +42,13 @@ export const processMultipleRepos = async (repos) => {
         data['createdAt'] = repo.createdAt
         data['updatedAt'] = repo.updatedAt
 
-        if ('resources' in datapackage) {
-          try {
-            data['schema'] = datapackage['resources'][0]['schema'] || []
-            data['resources'] = datapackage['resources']
-          } catch (error) {
-            data['schema'] = { fields: [] }
-            data['resources'] = datapackage['resources']
-          }
+        if ('resources' in datapackage && datapackage['resources'].length > 0) {
+          data['resources'] = datapackage['resources']
+          data['schema'] = datapackage['resources'][0]['schema'] || []
+          data['sample'] = datapackage['resources'][0]['sample']  || datapackage['sample'] || []
```

Comment:
@risenW, will we have a `sample` property at the top-level somewhere or is this to account for existing datasets which might have one already, as that's what seems to be the case from line 39 https://github.com/datopian/gift-portal/pull/164/files#diff-5dfe38baf287dcf756a17c2dd63483781b53bf4b669e10efdd01e74bcd8e780aR39
? (Thanks for enlightening me :smile:.)

```suggestion
          data['sample'] = datapackage['resources'][0]['sample'] || []
```

Reply:
@sglavoie Yes, this is for existing datasets with sample and schema in the top level. Should we not support it?